lib/eunit/src/eunit_proc.erl

```diff
@@ -273,7 +273,9 @@ kill_task(Child, St) ->
     terminate_insulator(St).
 
 buffer_to_binary([B]) when is_binary(B) -> B;  % avoid unnecessary copying
-buffer_to_binary(Buf) -> list_to_binary(lists:reverse(Buf)).
+buffer_to_binary(Buf) ->
```

Comment:
I don't see how this is meant to work. If you get a string with only characters from the Latin-1 subset of Unicode, it will get encoded as Latin-1 (exactly one byte per character, range 0-255), but if the string contains some code point above 255, you will generate a binary containing UTF-8 encoded characters instead. The code that eventually gets the binary will have no idea how to interpret it correctly.